Sources/GRPC/GRPCIdleHandler.swift

```diff
@@ -154,43 +171,58 @@ internal class GRPCIdleHandler: ChannelInboundHandler {
     let frame = self.unwrapInboundIn(data)
 
     if frame.streamID == .rootStream {
-      switch (self.state, frame.payload) {
-      // We only care about SETTINGS as long as we are in state `.notReady`.
-      case let (.notReady, .settings(content)):
-        self.state = .ready
-
-        switch self.mode {
-        case let .client(manager):
-          let remoteAddressDescription = context.channel.remoteAddress.map { "\($0)" } ?? "n/a"
-          manager.logger.info("gRPC connection ready", metadata: [
-            MetadataKey.remoteAddress: "\(remoteAddressDescription)",
-            MetadataKey.eventLoop: "\(context.eventLoop)",
-          ])
+      switch frame.payload {
+      case let .settings(.settings(settings)):
+        // Log any changes to HTTP/2 settings.
+        self.logger.debug(
+          "HTTP2 settings update",
+          metadata: Dictionary(settings.map {
+            ("\($0.parameter.loggingMetadataKey)", "\($0.value)")
+          }, uniquingKeysWith: { a, _ in a })
+        )
+
+        let maxConcurrentStreams = settings.first(where: { $0.parameter == .maxConcurrentStreams })
```

Comment:
A quick FYI: if there are multiple instances of a setting in a HTTP/2 settings frame, the _last_ instance of the setting is the one that counts. `HTTP2Settings` is a `RandomAccessCollection` so you can do `reversed()` for free.

Reply:
Oh, good catch. Thanks!